<?xml version="1.0" encoding="utf-8"?>
<launch>

  <group ns="robot">

    <!-- GDB functionality -->
    <arg name="debug" default="false" />
    <arg unless="$(arg debug)" name="launch_prefix" value="" />
    <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

    <!-- Load the URDF into the ROS Parameter Server -->
    <param name="robot_description"
	   command="cat '$(find baxter_description)/urdf/baxter.urdf'" />

    <!-- Load hardware interface -->
    <node name="baxter_hardware_interface" pkg="baxter_control" type="baxter_hardware_interface"
	  respawn="false" output="screen" launch-prefix="$(arg launch_prefix)" />

    <!-- Load joint controller configurations from YAML file to parameter server -->
    <rosparam file="$(find baxter_control)/config/hardware_controllers.yaml" command="load"/>

    <!-- Load the default controllers -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
	  output="screen" ns="/robot" args="right_joint_trajectory_controller
					    left_joint_trajectory_controller
					    " />
  </group>

  <!-- Load other configurations shared between all three Baxter run modes-->
  <include file="$(find baxter_control)/launch/baxter_configure.launch"/>    

</launch>